packages/oauth-helper/src/index.ts

```diff
@@ -0,0 +1,468 @@
+      await this.installationStore.storeInstallation(installation);
+      if (options != null && options.success != null) {
+        console.log('calling options.sucess');
+        options.success(installation, metadata, req, res);
+      } else {
+        // call our default callbackSuccess function
+        callbackSuccess(installation, metadata, req, res);
+      }
+    } catch (error) {
+      console.log(error);
+      if (options != null && options.failure != null) {
+        console.log('calling options.failure');
+        options.failure(error, this.installOptions , req, res);
+      } else {
+        console.log('run custom failure function');
+        callbackFailure(error, this.installOptions , req, res);
+      }
+    }
+  }
+}
+
+interface InstallProviderOptions {
+  clientId: string;
+  clientSecret: string;
+  stateStore?: StateStore; // default ClearStateStore
+  stateSecret?: string; // ClearStateStoreOptions['secret']; // required when using default stateStore
+  installationStore?: InstallationStore; // default MemoryInstallStore
+  authVersion: 'v1' | 'v2'; // default 'v2'
+}
+
+interface InstallURLOptions {
+  scopes: string | string[];
+  teamId?: string;
+  redirectUri?: string;
+  userScopes?: string | string[]; // cannot be used with authVersion=v1
+  metadata?: string;
+}
+
+interface CallbackOptions {
+  // success is given control after handleCallback() has stored the
+  // installation. when provided, this function must complete the
+  // callbackRes.
+  success?: (
+    installation: Installation,
+    metadata: string | undefined,
+    callbackReq: IncomingMessage,
+    callbackRes: ServerResponse,
+  ) => void;
+
+  // failure is given control when handleCallback() fails at any point.
+  // when provided, this function must complete the callbackRes.
+  // default:
+  // serve a generic "Error" web page (show detailed cause in development)
+  failure?: (
+    error: CodedError,
+    options: InstallURLOptions,
+    callbackReq: IncomingMessage,
+    callbackRes: ServerResponse,
+  ) => void;
+}
+
+// Response shape from oauth.v2.access - https://api.slack.com/methods/oauth.v2.access#response
+interface OAuthV2Response {
+  ok: boolean;
+  app_id: string;
+  authed_user: {
+    id: string,
+    scope?: string,
+    access_token?: string,
+    token_type?: string,
+  };
+  scope: string;
+  token_type: string;
+  access_token: string;
+  bot_user_id: string;
+  team: { id: string, name: string };
+  enterprise: { name: string, id: string } | null;
+  response_metadata: object;
+}
+
+// Response shape from oauth.access - https://api.slack.com/methods/oauth.access#response
+interface OAuthV1Response {
+  ok: boolean;
+  access_token: string;
+  // scope parameter isn't returned in workspace apps
+  scope: string;
+  team_name: string;
+  team_id: string;
+  enterprise_id: string | null;
+  // if they request bot user token
+  bot?: { bot_user_id: string, bot_access_token: string };
```

Comment:
I actually couldn't reproduce getting `bot` in my payload when using `oauth.access`. According to https://api.slack.com/docs/oauth#bots, all I need to do is add bot user and it should show up. Didn't work for me after I added the bot user. Not sure if there is a time delay or I need to request new scopes or something.

Reply:
hmm, still not seeing it. I'll try to debug more later to see if I can get it.